In [9]:
#importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

In [2]:
df=pd.read_csv('Week_8_Q&A_dataset - Sheet1.csv')

# 1. Carry out descriptive analytics on the dataset.

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Gender            28 non-null     object
 1   Test preparation  28 non-null     object
 2   Total Marks       28 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 800.0+ bytes


In [5]:
df.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

1.The given dataset contains 28 samples and 3 features.

2.There is zero null values in this dataset.

3.The average marks scored is 32.32 with a standard deviation of 12.45.

# 2. Check whether female and male students scored the same marks.

Hypothesis:

Ho:The average score of female and male students is equal

Ha:The average scores are not equal

alpha: 0.05

Hence we are performing two-tailed t-test on the dataset to analyse our hypothesis

In [6]:
df1=df.groupby('Gender')
male = df.query('Gender == "male"')['Total Marks']
female = df.query('Gender == "female"')['Total Marks']

In [21]:
tvalue, pvalue=stats.ttest_ind(male,female)
print("The t value of means is: ", round(tvalue, 4))
print("The p value based on the t score is: ", round(pvalue,4))

The t value of means is:  -0.6941
The p value based on the t score is:  0.4938


As the p-value=0.4938 > 0.05 and hence we cannot reject the null Hypothesis
ie. Female and Male students scored the same marks

# 3.Whether test preparation helps the students?

Hypothesis:

Ho :The average score of students who completed vs students who didn't is equal

Ha :The average scores of students who completed are higher

alpha: 0.05

Hence performing one-tailed t-test on the dataset to analyse our hypothesis

In [18]:
df2=df.groupby('Test preparation')
completed=df[df['Test preparation'] =="completed"]['Total Marks']
none=df[df['Test preparation'] =="none"]['Total Marks']

In [22]:
tvalue, pvalue=stats.ttest_ind(completed,none)
print("The t value of means is: ", round(tvalue, 4))
print("The p value based on the t score is: ", round(pvalue,4))

The t value of means is:  3.6144
The p value based on the t score is:  0.0013


As the p-value 0.0013 < 0.05, we reject the null hypothesis.
ie. We can say that Average marks for students completed the test is higher .